api/memory.go

```diff
@@ -29,58 +29,49 @@ func makeView(s []byte) C.ByteSliceView {
 	}
 }
 
-func allocateRust(data []byte) C.Buffer {
-	var ret C.Buffer
+func newUnmanagedVector(data []byte) C.UnmanagedVector {
```

Comment:
This name is more confusing than the old one.

`copyToRust` or something would be clear that we copy any existing Go data into a Rust Vector (to be consumed there)

Reply:
Yeah, copy is a good point. I think it is importent to make explicit we create a new `UnmanagedVector` because this we have to do manually for those: create ad destroy.